pkg/service/controller.go

```diff
@@ -52,6 +70,25 @@ func (c *ControllerService) validateCreateVolumeRequest(req *csi.CreateVolumeReq
 		return status.Error(codes.InvalidArgument, "volume capabilities missing in request")
 	}
 
+	csiConfigMap, err := c.infraClientset.CoreV1().ConfigMaps(c.infraClusterNamespace).Get(ctx, configMapName, metav1.GetOptions{})
+	if err != nil {
+		return status.Error(codes.FailedPrecondition, "Couldn't find a configmap for the csi driver")
+	}
+
+	var allowedStorageClassesArr []string
+	allowedStorageClasses, ok := csiConfigMap.Data[allowedInfraStorageclassesKey] //Todo: check if key exists
+	if !ok {
+		allowedStorageClassesArr = defaultAllowedInfraStorageclasses
+	} else {
+		allowedStorageClassesArr = strings.Split(allowedStorageClasses, ",")
+	}
+
+	storageClassName := req.Parameters[InfraStorageClassNameParameter]
+
+	if !contains(allowedStorageClassesArr, storageClassName) {
+		return status.Error(codes.InvalidArgument, "infraStorageclass is not in the allowed list")
+	}
+
```

Comment:
you don't need to make an API call to get the configMap. we can pass in this `allowedStorageClassArg` into the deployment using an env var like we do everything else in the configmap. example, https://github.com/kubevirt/csi-driver/blob/4afb019ce9f71b0cdc93c2ae44a35332064b383a/deploy/040-controller.yaml#L54